modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/VerticalCardTag.java

```diff
@@ -0,0 +1,821 @@
+	 */
+	@Deprecated
+	@Override
+	public Map<String, String> getData() {
+		if ((super.getData() == null) && (_verticalCard != null)) {
+			return _verticalCard.getData();
+		}
+
+		return super.getData();
+	}
+
+	/**
+	 * @deprecated As of Cavanaugh (7.4.x), with no direct replacement
+	 */
+	@Deprecated
+	@Override
+	public String getDefaultEventHandler() {
+		if ((super.getDefaultEventHandler() == null) &&
+			(_verticalCard != null)) {
+
+			return _verticalCard.getDefaultEventHandler();
+		}
+
+		return super.getDefaultEventHandler();
+	}
+
+	public Boolean getDisabled() {
+		if (_disabled == null) {
+			if (_verticalCard != null) {
+				return _verticalCard.getDisabled();
+			}
+
+			return false;
+		}
+
+		return _disabled;
+	}
+
+	/**
+	 * @deprecated As of Cavanaugh (7.4.x), replaced by {@link #getCssClass()}
+	 */
+	@Deprecated
+	@Override
+	public String getElementClasses() {
+		if ((super.getCssClass() == null) && (_verticalCard != null)) {
+			return _verticalCard.getElementClasses();
+		}
+
+		return super.getCssClass();
+	}
+
+	public Boolean getFlushHorizontal() {
+		if (_flushHorizontal == null) {
+			if (_verticalCard != null) {
+				return _verticalCard.getFlushHorizontal();
+			}
+
+			return false;
+		}
+
+		return _flushHorizontal;
+	}
+
+	public Boolean getFlushVertical() {
+		if (_flushVertical == null) {
+			if (_verticalCard != null) {
+				return _verticalCard.getFlushVertical();
+			}
+
+			return false;
+		}
+
+		return _flushVertical;
+	}
+
+	/**
+	 * @deprecated As of Cavanaugh (7.4.x), with no direct replacement
+	 */
+	@Deprecated
+	public String getGroupName() {
+		return _groupName;
+	}
+
+	public String getHref() {
+		if ((_href == null) && (_verticalCard != null)) {
+			return _verticalCard.getHref();
+		}
+
+		return _href;
+	}
+
+	public String getIcon() {
+		if ((_icon == null) && (_verticalCard != null)) {
+			return _verticalCard.getIcon();
+		}
+
+		return _icon;
+	}
+
+	@Override
+	public String getId() {
+		if ((super.getId() == null) && (_verticalCard != null)) {
+			return _verticalCard.getId();
+		}
+
+		return super.getId();
+	}
+
+	public String getImageAlt() {
+		if ((_imageAlt == null) && (_verticalCard != null)) {
+			return _verticalCard.getImageAlt();
+		}
+
+		return _imageAlt;
+	}
+
+	public String getImageSrc() {
+		if ((_imageSrc == null) && (_verticalCard != null)) {
+			return _verticalCard.getImageSrc();
+		}
+
+		return _imageSrc;
+	}
+
+	public String getInputName() {
+		if ((_inputName == null) && (_verticalCard != null)) {
+			return _verticalCard.getInputName();
+		}
+
+		return _inputName;
+	}
+
+	public String getInputValue() {
+		if ((_inputValue == null) && (_verticalCard != null)) {
+			return _verticalCard.getInputValue();
+		}
+
+		return _inputValue;
+	}
+
+	public List<LabelItem> getLabels() {
+		if ((_labels == null) && (_verticalCard != null)) {
+			return _verticalCard.getLabels();
+		}
+
+		return _labels;
+	}
+
+	public Map<String, String> getLabelStylesMap() {
```

Comment:
I missed all these comments - sorry about that

You're right.


I created [this issue](https://github.com/liferay/clay/issues/3777) in the clay repository to talk about it and add it to clay 3.x if required. If the decision is to stop using this I'll deprecate.

Reply:
So I looked for documentation about `labelStylesMap` with `git grep -w labelStylesMap` in the clay repository but didn't find anything very interesting.

The best information I could find was [this](https://github.com/liferay/clay/blob/cd994a00f73063d22495f295c00f05c736ef6005/packages/clay-card/src/ClayCardBase.js#L207-L214)

I'm a bit surprised because when using `<clay:vertical-card-v2>` (i.e. clay 2.x) this doesn't seem to work, at least in the `frontend-taglib-clay-sample-web` 

```jsp
<clay:image-card-v2
	actionDropdownItems="<%= claySampleImageCard.getActionDropdownItems() %>"
	icon="camera"
	labels="<%= claySampleImageCard.getLabels() %>"
	labelStylesMap="<%= claySampleImageCard.getLabelStylesMap() %>"
	stickerLabel="SVG"
	stickerStyle="warning"
	subtitle="<%= claySampleImageCard.getSubtitle() %>"
	title="<%= _SVG_FILE_TITLE %>"
/>
```

```java
// import declarations

public class ClaySampleImageCard implements ImageCard {

	// more methods ...

	public List<LabelItem> getLabels() {
		if (_labels != null) {
			return _labels;
		}

		int numItems = 1 + RandomUtil.nextInt(3);

		return LabelItemListBuilder.add(
			labelItem -> {
				labelItem.setLabel("Approved");
				labelItem.setStyle("success");
			}
		).add(
			() -> numItems > 1, labelItem -> labelItem.setLabel("Pending")
		).add(
			() -> numItems > 2,
			labelItem -> {
				labelItem.setLabel("Canceled");
				labelItem.setStyle("danger");
			}
		).build();
	}

	public Map<String, String> getLabelStylesMap() {
		if (_labelStylesMap != null) {
			return _labelStylesMap;
		}

		_labelStylesMap = HashMapBuilder.put(
			"Pending", "warning"
		).build();

		return _labelStylesMap;
	}

	// more methods ...
}
```

Result:

![](https://user-images.githubusercontent.com/5572/98251497-0f1d1480-1f79-11eb-8138-7dad116facae.png)